### PR TITLE
Pipeline pgcraft misc

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/backends/gl/pipeline_gl.c [hunk 1]
```diff
@@ -52,6 +52,8 @@ struct buffer_binding {
     GLuint type;
     struct pipeline_buffer_desc desc;
     const struct buffer *buffer;
+    size_t offset;
+    size_t size;
```
Comment (ubitux) [inline libnopegl/src/backends/gl/pipeline_gl.c:56]: does/do typo in commit message

File: libnopegl/src/pipeline.h [hunk 2]
```diff
@@ -64,8 +64,6 @@ struct pipeline_buffer_desc {
     int binding;
     int access;
     int stage;
-    size_t offset;
-    size_t size;
```
Comment (ubitux) [inline libnopegl/src/pipeline.h:68]: Is this also true for the `offset` of the pipeline attribute desc?

File: libnopegl/src/pipeline.h [hunk 3]
```diff
@@ -52,7 +52,6 @@ struct pipeline_uniform_desc {
 struct pipeline_texture_desc {
     char name[MAX_ID_LEN];
     int type;
-    int location;
```
Comment (ubitux) [inline libnopegl/src/pipeline.h:55]: So only the location of the attributes remain in the pipeline descriptors; are those still necessary?

File: libnopegl/src/pipeline.h (hunk 2 above)
Comment (mbouron) [reply libnopegl/src/pipeline.h:68]: No.

File: libnopegl/src/pipeline.h (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/pipeline.h:55]: It is. The terminology used for attribute is location not binding.

File: libnopegl/src/backends/gl/pipeline_gl.c (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/backends/gl/pipeline_gl.c:56]: Done.

File: libnopegl/src/backends/vk/gpu_ctx_vk.c [hunk 4]
```diff
@@ -905,6 +905,7 @@ static int vk_init(struct gpu_ctx *s)
     /* max_texture_image_units is specific to the OpenGL backend and has no
      * direct Vulkan equivalent so use a sane default value */
     s->limits.max_texture_image_units            = 32;
+    s->limits.max_image_units                    = 32;
```
[the remark this answers is not in the document]
Comment (mbouron) [reply libnopegl/src/backends/vk/gpu_ctx_vk.c:908]: Done.

**Review by ubitux — APPROVED** [review]

File: libnopegl/src/backends/gl/pipeline_gl.c [hunk 5]
```diff
@@ -204,26 +207,18 @@ static int build_texture_bindings(struct pipeline *s)
             texture_desc->type == NGLI_TYPE_IMAGE_CUBE) {
             struct gpu_ctx_gl *gpu_ctx_gl = (struct gpu_ctx_gl *)s->gpu_ctx;
             struct glcontext *gl = gpu_ctx_gl->glcontext;
-            const struct gpu_limits *limits = &gl->limits;
 
             if (!(gl->features & NGLI_FEATURE_GL_SHADER_IMAGE_LOAD_STORE)) {
                 LOG(ERROR, "context does not support shader image load store operations");
                 return NGL_ERROR_GRAPHICS_UNSUPPORTED;
             }
```
[the remark this answers is not in the document]
Comment (mbouron) [reply libnopegl/src/backends/gl/pipeline_gl.c]: Done in a separate commit.

**Pull request merged**